### DISCUSSION TITLE
ApplicationSets vs Projects

### DISCUSSION THREAD

**Discussion by Shubham1406220** [argoproj/argo-cd#12112] [General]
I have started with using argoCD but i couldn't find the difference between Project and applicationSets. I want to know which option will be better for the scenario.

We have 3 environment and we are using 1 argoCD to deploy apps to all 3 environments. We were thinking of setting up 3 projects with required privileges and deploy the apps from the project. How applicationset will come into the picture in this setup.

**Comment by crenshaw-dev** [comment]
Projects determine permissions. ApplicationSets just automate creation of Applications. Each generated Application can use a different project or the same project, depending on how you template the ApplicationSet.

I do recommend having three Projects. It’s good to enforce least necessary privileges for each environment.

Whether you use ApplicationSets depends completely on whether you need the automation. It’s a completely separate question from how many Projects you need.

**Reply by Shubham1406220** [reply]
Thanks for helping @crenshaw-dev
